src/lib/pubkey/dilithium/dilithium_polynomials.h

```diff
@@ -0,0 +1,1586 @@
+         {
+            for (size_t i = 0; i < DilithiumModeConstants::N / 2; ++i)
+            {
+               t[0] = static_cast<uint32_t>(mode.gamma1()) - m_coeffs[2 * i + 0];
+               t[1] = static_cast<uint32_t>(mode.gamma1()) - m_coeffs[2 * i + 1];
+
+               r[5 * i + 0] = static_cast<uint8_t>(t[0]);
+               r[5 * i + 1] = static_cast<uint8_t>(t[0] >> 8);
+               r[5 * i + 2] = static_cast<uint8_t>(t[0] >> 16);
+               r[5 * i + 2] |= static_cast<uint8_t>(t[1] << 4);
+               r[5 * i + 3] = static_cast<uint8_t>(t[1] >> 4);
+               r[5 * i + 4] = static_cast<uint8_t>(t[1] >> 12);
+            }
+         }
+      }
+
+      /*************************************************
+      * Name:        polyt1_unpack
+      *
+      * Description: Unpack polynomial t1 with 10-bit coefficients.
+      *              Output coefficients are standard representatives.
+      *
+      * Arguments:   - Polynomial& r: pointer to output polynomial
+      *              - const uint8_t *a: byte array with bit-packed_t1 polynomial
+      **************************************************/
+      static void polyt1_unpack(Polynomial& r, const uint8_t* a)
+      {
+         for (size_t i = 0; i < DilithiumModeConstants::N / 4; ++i)
+         {
+            r.m_coeffs[4 * i + 0] = ((a[5 * i + 0] >> 0) | (static_cast<uint32_t>(a[5 * i + 1]) << 8)) & 0x3FF;
+            r.m_coeffs[4 * i + 1] = ((a[5 * i + 1] >> 2) | (static_cast<uint32_t>(a[5 * i + 2]) << 6)) & 0x3FF;
+            r.m_coeffs[4 * i + 2] = ((a[5 * i + 2] >> 4) | (static_cast<uint32_t>(a[5 * i + 3]) << 4)) & 0x3FF;
+            r.m_coeffs[4 * i + 3] = ((a[5 * i + 3] >> 6) | (static_cast<uint32_t>(a[5 * i + 4]) << 2)) & 0x3FF;
+         }
+      }
+
+      /*************************************************
+      * Name:        polyt1_pack
+      *
+      * Description: Bit-pack polynomial t1 with coefficients fitting in 10 bits.
+      *              Input coefficients are assumed to be standard representatives.
+      *
+      * Arguments:   - uint8_t *r: pointer to output byte array with at least
+      *                            POLYT1_PACKEDBYTES bytes
+      **************************************************/
+      void polyt1_pack(uint8_t *r) const
+      {
+         for (size_t i = 0; i < DilithiumModeConstants::N / 4; ++i)
+         {
+            r[5 * i + 0] = static_cast<uint8_t>((m_coeffs[4 * i + 0] >> 0));
+            r[5 * i + 1] = static_cast<uint8_t>((m_coeffs[4 * i + 0] >> 8) | (m_coeffs[4 * i + 1] << 2));
+            r[5 * i + 2] = static_cast<uint8_t>((m_coeffs[4 * i + 1] >> 6) | (m_coeffs[4 * i + 2] << 4));
+            r[5 * i + 3] = static_cast<uint8_t>((m_coeffs[4 * i + 2] >> 4) | (m_coeffs[4 * i + 3] << 6));
+            r[5 * i + 4] = static_cast<uint8_t>((m_coeffs[4 * i + 3] >> 2));
+         }
+      }
+
+      Polynomial() = default;
+   };
+   class PolynomialVector
```

Comment:
This class (and potentially others) are clashing with the Kyber implementation (that also contains a `PolynomialVector`) when building as an amalgamation. In that case, `./configure.py` mashes all headers and source files into the same compilation unit. In that case, `PolynomialVector` is defined twice resulting in a build failure.

For the time being we should probably just namespace those helpers, like: `Botan::Dilithium::PolynomialVector`. In the long run, I would hope that we can abstract those classes to be usable for both Kyber and Dilithium.